integration-tests/modules/__tests__/product/store/index.spec.ts

```diff
@@ -0,0 +1,332 @@
+      let productModule: IProductModuleService
+      let salesChannelModule: ISalesChannelModuleService
+
+      beforeAll(async () => {
+        appContainer = getContainer()
+        pricingModule = appContainer.resolve(ModuleRegistrationName.PRICING)
+        productModule = appContainer.resolve(ModuleRegistrationName.PRODUCT)
+        salesChannelModule = appContainer.resolve(
+          ModuleRegistrationName.SALES_CHANNEL
+        )
+      })
+
+      beforeEach(async () => {
+        await createAdminUser(dbConnection, adminHeaders, appContainer)
+        await createDefaultRuleTypes(appContainer)
+      })
+
+      describe("GET /store/products", () => {
+        beforeEach(async () => {
+          ;[product, [variant]] = await createProductsWithVariants(
+            productModule,
+            {
+              title: "test product 1",
+              status: ProductStatus.PUBLISHED,
+              variants: [{ title: "test variant 1" }],
+            }
+          )
+          ;[product2, [variant2]] = await createProductsWithVariants(
+            productModule,
+            {
+              title: "test product 2 uniquely",
+              status: ProductStatus.PUBLISHED,
+              variants: [{ title: "test variant 2" }],
+            }
+          )
+          ;[product3, [variant3]] = await createProductsWithVariants(
+            productModule,
+            {
+              title: "product not in price list",
+              status: ProductStatus.PUBLISHED,
+              variants: [{ title: "test variant 3" }],
+            }
+          )
+          ;[product4, [variant4]] = await createProductsWithVariants(
+            productModule,
+            {
+              title: "draft product",
+              status: ProductStatus.DRAFT,
+              variants: [{ title: "test variant 4" }],
+            }
+          )
+        })
+
+        it("should list all published products", async () => {
+          let response = await api.get(`/store/products`)
+
+          expect(response.status).toEqual(200)
+          expect(response.data.count).toEqual(3)
+          expect(response.data.products).toEqual(
+            expect.arrayContaining([
+              expect.objectContaining({
+                id: product.id,
+              }),
+              expect.objectContaining({
+                id: product2.id,
+              }),
+              expect.objectContaining({
+                id: product3.id,
+              }),
+            ])
+          )
+
+          response = await api.get(`/store/products?q=uniquely`)
+
+          expect(response.status).toEqual(200)
+          expect(response.data.count).toEqual(1)
+          expect(response.data.products).toEqual([
+            expect.objectContaining({
+              id: product2.id,
+            }),
+          ])
+        })
+
+        it("should list all products for a sales channel", async () => {
+          const salesChannel = await salesChannelModule.create({
```

Comment:
If you use the HTTP admin endpoints you won't need to do any of the remote linking, you'll simply create the salesChannel and then pass the sales_channel_id to the createProduct call